src/main/java/uk/gov/companieshouse/api/accounts/CompanyAccountsApplication.java

```diff
@@ -1,57 +1,71 @@
 package uk.gov.companieshouse.api.accounts;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Primary;
 import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import uk.gov.companieshouse.api.accounts.interceptor.CompanyAccountInterceptor;
+import uk.gov.companieshouse.api.accounts.interceptor.SmallFullInterceptor;
 import uk.gov.companieshouse.api.accounts.interceptor.TransactionInterceptor;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
 
 @SpringBootApplication
 public class CompanyAccountsApplication implements WebMvcConfigurer {
 
-  public static final String APPLICATION_NAME_SPACE = "company-accounts.api.ch.gov.uk";
-  private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+    public static final String APPLICATION_NAME_SPACE = "company-accounts.api.ch.gov.uk";
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @Autowired
+    private SmallFullInterceptor smallFullInterceptor;
+
+    @Autowired
+    private CompanyAccountInterceptor companyAccountInterceptor;
+
+    @Autowired
+    private TransactionInterceptor transactionInterceptor;
 
-  @Autowired
-  private TransactionInterceptor transactionInterceptor;
+    public static void main(String[] args) {
 
-  public static void main(String[] args) {
+        // Returns the configured ${PORT} value passed in under `server.port`.
+        // If no port is configured, return null
+        Integer port = Integer.getInteger("server.port");
+
+        if (port == null) {
+            LOGGER.error("Failed to start service, no port has been configured");
+            System.exit(0);
+        }
+
+        SpringApplication.run(CompanyAccountsApplication.class, args);
+    }
+
+    @Override
+    public void addInterceptors(final InterceptorRegistry registry) {
+        registry.addInterceptor(transactionInterceptor)
+                .addPathPatterns("/transactions/{transactionId}/**");
+        registry.addInterceptor(companyAccountInterceptor)
+                .addPathPatterns(
+                        "/transactions/{transactionId}/company-accounts/{companyAccountId}/**");
```

Comment:
This should match `/transactions/{transactionId}/company-accounts/{companyAccountId}**` - notice the last slash has been removed.

This is to ensure that this interceptor is run against the GET, PUT and DELETE methods (not POST which doesn't have the companyAccountId) of `company-accounts` i.e. the check that ensures that the company-accounts resource is on the transaction referenced in the URL

This also gives us the GET implementation for free as if it passes the interceptor we will already have the company-accounts data in the session so the controller can just return that data from the session